context.ts

```diff
@@ -0,0 +1,49 @@
+import type { DecoManifest, LiveOptions } from "./types.ts";
+
+type EnhancedLiveOptions = LiveOptions & { siteId?: number };
+
+// While Fresh doesn't allow for injecting routes and middlewares,
+// we have to deliberately store the manifest in this scope.
+let manifest: DecoManifest;
+let liveOptions: EnhancedLiveOptions;
+const defaultDomains = [
+  `localhost`,
+];
+const deploymentId = Deno.env.get("DENO_DEPLOYMENT_ID");
+const _isDenoDeploy = deploymentId !== undefined;
+
+export function getDefaultDomains() {
+  return defaultDomains;
+}
+
+export function pushDefaultDomains(...domains: string[]) {
+  defaultDomains.push(...domains);
+}
+
+export function isPrivateDomain(domain: string) {
+  return defaultDomains.includes(domain);
+}
+
+export function getManifest() {
+  return manifest;
+}
+
+export function setManifest(newManifest: DecoManifest) {
+  manifest = newManifest;
+}
+
+export function getLiveOptions() {
+  return liveOptions;
+}
+
+export function setLiveOptions(newLiveOptions: EnhancedLiveOptions) {
+  liveOptions = newLiveOptions;
+}
+
+export function getDeploymentId() {
+  return deploymentId;
+}
+
+export function isDenoDeploy() {
+  return _isDenoDeploy;
+}
```

Comment:
I don't usually use classes, but this one might be a good fit:

- stateful
- offers an interface to get/set
- singleton